src/app/core/auth.service.ts

```diff
@@ -0,0 +1,62 @@
+import { AngularFirestore } from '@angular/fire/firestore';
+import { AppUser } from './../models/user.interface';
+import { Injectable } from '@angular/core';
+import { AngularFireAuth } from '@angular/fire/auth';
+import { switchMap } from 'rxjs/operators';
+import { Observable, of } from 'rxjs';
+import { auth, User } from 'firebase/app';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  user$: Observable<AppUser>;
+
+  constructor(private afAuth: AngularFireAuth, private db: AngularFirestore) {
+    this.user$ = this.afAuth.authState.pipe(
+      switchMap((fbUser) => {
+        if (fbUser) {
+          return this.db
+            .collection('users')
+            .doc<AppUser>(fbUser.uid)
+            .valueChanges();
+        } else {
+          return of(null);
+        }
+      })
+    );
+  }
+
+  googleLogin() {
+    this.afAuth
+      .signInWithPopup(new auth.GoogleAuthProvider())
+      .then((credential) => this.updateUser(credential.user));
+  }
+
+  logOut() {
+    this.afAuth.signOut();
+  }
+
+  private updateUser(user: User) {
+    this.db
+      .collection('users')
+      .doc<AppUser>(user.uid)
+      .set({
+        email: user.email,
+        displayName: user.displayName,
+        photoURL: user.photoURL,
+        rol: {
+          user: true,
+          admin: this.isAdmin(user.email),
+        },
+      });
+  }
+
+  private isAdmin(email: string) {
+    return Boolean(
```

Comment:
Just wondering in case I'm missing something, why do you need the `Boolean` constructor here?

Also, I think the boolean expression needs adjustment. The function will always return true, it should be something like:

```
email === 'nico' || email === 'carlos' || email === 'juan'
```

As this can become verbose with more admins you could write a whitelist instead

```typescript
function isAdmin(email: string) {
  const admins = ['nico', 'carlos', 'juan']
  return admins.indexOf(email) !== -1;
}
```